packages/camera/camera_avfoundation/example/ios/RunnerTests/FLTCamPhotoCaptureTests.m

```diff
@@ -94,4 +94,86 @@ - (void)testCaptureToFile_mustReportPathToResultIfSavePhotoDelegateCompletionsWi
   [self waitForExpectationsWithTimeout:1 handler:nil];
 }
 
+- (void)testCaptureToFile_mustSetExtensionToHeicIfAvailablePhotoCodecTypesContainsHEVC {
+  XCTestExpectation *expectation =
+      [self expectationWithDescription:
+                @"Test must set extension to heif if availablePhotoCodecTypes contains HEVC."];
+  dispatch_queue_t captureSessionQueue = dispatch_queue_create("capture_session_queue", NULL);
+  dispatch_queue_set_specific(captureSessionQueue, FLTCaptureSessionQueueSpecific,
+                              (void *)FLTCaptureSessionQueueSpecific, NULL);
+  FLTCam *cam = FLTCreateCamWithCaptureSessionQueue(captureSessionQueue);
+  [cam setFileFormat:FCPFileFormatHEIF];
+
+  AVCapturePhotoSettings *settings =
+      [AVCapturePhotoSettings photoSettingsWithFormat:@{AVVideoCodecKey : AVVideoCodecTypeHEVC}];
+
+  id mockSettings = OCMClassMock([AVCapturePhotoSettings class]);
+  OCMStub([mockSettings photoSettingsWithFormat:OCMOCK_ANY]).andReturn(settings);
+
+  NSString *filePath = @"test";
+  id mockResult = OCMClassMock([FLTThreadSafeFlutterResult class]);
+  OCMStub([mockResult sendSuccessWithData:filePath]).andDo(^(NSInvocation *invocation) {
+    [expectation fulfill];
```

Comment:
the expectation is `"Test must set extension to heif if availablePhotoCodecTypes contains HEVC."`, but here it fulfilled without verifying the extension.